File: ExcelDivider/src/main/java/ru/tsystems/divider/service/api/JiraToDBConverter.java
package ru.tsystems.divider.service.api;

import java.io.IOException;

import ru.tsystems.divider.exceptions.NoShetException;

public interface JiraToDBConverter {

    /**
     * Read excel file which created by jira and write it to ad DB. For xlsx files only
     *
     * @param fileName
     *            Excel file name.
     * @param sheetName
     *            Excel sheet name with jira columns.
     * @param startRowIndex
     *            First row index with needed data.
     */
    void transferAll(String fileName, String sheetName, int startRowIndex)throws IOException, NoShetException;

    /**
     * Read excel file which created by jira and write it to ad DB. For xls files only
     *
     * @param fileName
     *            Excel file name.
     * @param sheetName
     *            Excel sheet name with jira columns.
     * @param startRowIndex
     *            First row index with needed data.
     */
    void transferAllOldExcell(String fileName, String sheetName, int startRowIndex)throws IOException, NoShetException;
    void transferAllCommentsOldExcell(String fileName, String sheetName, int startRowIndex)throws IOException, NoShetException;
    void transferAllComments(String fileName, String sheetName, int startRowIndex)throws IOException, NoShetException;
}

File: ExcelDivider/src/main/kt/ru/tsystems/divider/service/api/JiraToDBConverter.kt
package ru.tsystems.divider.service.api

import ru.tsystems.divider.exceptions.NoShetException
import java.io.IOException


//todo: deprecated "OldExcel" methods, if it possible
interface JiraToDBConverter {

    /**
     * Read excel file which created by jira and write it to ad DB. For xlsx files only
     *
     * @param fileName
     * Excel file name.
     * @param sheetName
     * Excel sheet name with jira columns.
     * @param startRowIndex
     * First row index with needed data.
     */
    @Throws(IOException::class, NoShetException::class)
    fun transferAll(excelReader: ExcelFileReader, startRowIndex: Int)

    @Throws(IOException::class, NoShetException::class)
    fun transferAllComments(excelReader: ExcelFileReader, startRowIndex: Int)
}
